include/rocksdb/slice.h

```diff
@@ -56,6 +56,7 @@ class Slice {
 
   // Return the length (in bytes) of the referenced data
   size_t size() const { return size_; }
+  inline void set_size(size_t s) { size_ = s; }
```

Comment:
Is there a way to avoid this hack? Like define another variable in IndexBlockIter? `Slice` is a public facing class and we are introducing a way to break its integrity.